x/batch/keeper/delegation.go

```diff
@@ -0,0 +1,130 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// increment the reference count for a historical rewards value
```

Comment:
can we reference where these are copied from

Reply:
I think it was copied from here.
https://github.com/cosmos/cosmos-sdk/blob/main/x/distribution/keeper/validator.go#L122C11-L122C54